soc/xtensa/espressif_esp32/common/Kconfig.soc

```diff
@@ -37,6 +37,13 @@ config ESP_HEAP_SEARCH_ALL_REGIONS
 	  regions. If the region of desired capability is exhausted,
 	  memory will be allocated from other available region.
 
+config SPIRAM_HEAP_SIZE
+	int "Size of SPIRAM heap"
+	default 1048576
```

Comment:
I think this default value will fail the assertion in `sys_heap_init` (heap.c) for `esp32_devkitc_wrover` and other boards based on ESP32. It has the chosen node `zephyr,sram = &sram0` with size `0x2c200`. This small size enables `CONFIG_SYS_HEAP_SMALL_ONLY` that allows the maximum heap size of `0x3FF8`. I haven't tested it though, did you? If it fails to init this heap I suggest using default value `262134` for `SOC_SERIES_ESP32` until we figure the best way to dodge this issue.

Reply:
Thank you, that is a good catch.  I did a build test for `esp32_devkitc_wrover`, but didn't have an ESP32 with SPIRAM available to do a physical test.  I'll dig around in my storage boxes since I should have an WT32-SC01 which has SPIRAM.

Reply:
The sys_heap_init() __ASSERT() wasn't triggered at build time, but I confirmed that the ESP32 did not finishing booting on the WT32-SC01 which uses an ESP32-WROVER-B-N4R8 module.  No debug messages when it failed to boot and enabling a minimal core dump also didn't print anything presumably because it is too early in init.  I don't have a JTAG connector for the board, so I wasn't able to confirm it was the assertion.

However, changing the default size to 262134 if CONFIG_SYS_HEAP_SMALL_ONLY is enabled did allow it to boot and the test was successful.

I have updated the Kconfig to the following and will push it tonight.

```c
config SPIRAM_HEAP_SIZE
	int "Size of SPIRAM heap"
	default 262134 if SYS_HEAP_SMALL_ONLY
	default 1048576 if !SYS_HEAP_SMALL_ONLY
	depends on ESP_SPIRAM
	help
	  Specify size of SPIRAM heap.
```

Using `west build -b esp32_devkitc_wrover tests/boards/espressif_esp32/cache_coex`, the test run on an ESP32-WROVER-B-N4R8:
```c
Jun  8 2016 00:22:57
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:6608
load:0x40078000,len:15224
load:0x40080400,len:3992
entry 0x40080634
I (27) boot: ESP-IDF e2a4f32fb-dirty 2nd stage bootloader
I (27) boot: compile time 14:18:04
I (27) boot: chip revision: 1
I (30) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (47) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (57) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (75) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (82) boot:  2 factory          factory app      00 00 00010000 00100000
I (90) boot: End of partition table
I (94) boot_comm: chip revision: 1, min. application chip revision: 0
I (101) esp_image: segment 0: paddr=00010020 vaddr=00000020 size=0001ch (    28) 
I (110) esp_image: segment 1: paddr=00010044 vaddr=3ffb0000 size=00998h (  2456) load
I (119) esp_image: segment 2: paddr=000109e4 vaddr=3ffb0998 size=00194h (   404) load
I (127) esp_image: segment 3: paddr=00010b80 vaddr=40080000 size=06c2ch ( 27692) load
I (146) esp_image: segment 4: paddr=000177b4 vaddr=00000000 size=08884h ( 34948) 
I (159) esp_image: segment 5: paddr=00020040 vaddr=3f400040 size=02b5ch ( 11100) map
I (164) esp_image: segment 6: paddr=00022ba4 vaddr=00000000 size=0d474h ( 54388) 
I (185) esp_image: segment 7: paddr=00030020 vaddr=400d0020 size=055d4h ( 21972) map
I (196) boot: Loaded app from partition at offset 0x10000
I (196) boot: Disabling RNG early entropy source...
I (231) psram: This chip is ESP32-D0WD
I (232) spiram: Found 64MBit SPI RAM device
I (232) spiram: SPI RAM mode: flash 80m sram 40m
 *** Booting Zephyr OS build zephyr-v3.4.0-1825-ga1e9f697012a ***de.
Running TESTSUITE cache_coex
===================================================================
START - test_flash_integrity
 PASS - test_flash_integrity in 0.001 seconds
===================================================================
START - test_ram_integrity
 PASS - test_ram_integrity in 0.001 seconds
===================================================================
START - test_using_spiram
 PASS - test_using_spiram in 0.001 seconds
===================================================================
TESTSUITE cache_coex succeeded
------ TESTSUITE SUMMARY START ------
SUITE PASS - 100.00% [cache_coex]: pass = 3, fail = 0, skip = 0, total = 3 duration = 0.003 seconds
 - PASS - [cache_coex.test_flash_integrity] duration = 0.001 seconds
 - PASS - [cache_coex.test_ram_integrity] duration = 0.001 seconds
 - PASS - [cache_coex.test_using_spiram] duration = 0.001 seconds
------ TESTSUITE SUMMARY END ------
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

Reply:
Heap size configuration options pushed.

Are you able to run the `west build -b esp32_devkitc_wrover tests/boards/espressif_esp32/cache_coex` test on an ESP32-S2?